lms/djangoapps/courseware/tests/test_views.py

```diff
@@ -1055,6 +1055,21 @@ def test_query_counts(self, (sql_calls, mongo_calls, self_paced), self_paced_ena
             resp = views.progress(self.request, course_id=unicode(self.course.id))
         self.assertEqual(resp.status_code, 200)
 
+    @patch('courseware.grades.grade', Mock(return_value={'grade': 'Pass', 'percent': 0.75, 'section_breakdown': [],
+                                                         'grade_breakdown': []}))
+    def test_generate_cert_for_mode(self):
+        CertificateGenerationConfiguration(enabled=True).save()
+        certs_api.set_cert_generation_enabled(self.course.id, True)
+        CourseEnrollment.enroll(self.user, self.course.id, mode=CourseMode.AUDIT)
+        resp = views.progress(self.request, course_id=unicode(self.course.id))
+        # Check that the progress page has no 'Request Certificate' button for audit.
+        self.assertNotContains(resp, 'Request Certificate')
+        # Update the user enrollment to verified
+        CourseEnrollment.enroll(self.user, self.course.id, mode=CourseMode.VERIFIED)
+        resp = views.progress(self.request, course_id=unicode(self.course.id))
+        # Check that the progress page has 'Request Certificate' button for non audit.
+        self.assertContains(resp, 'Request Certificate')
```

Comment:
this is a good test — I'm glad you checked an audit and non-audit student in the same test :)